New traitor item: Compact Bow.

Xackii opened the pull request:
## About The Pull Request

Add new traitor item - Compact Bow. cost 11 tc.

https://youtu.be/IhoP98mc87E

The compact bow is a large item that cannot be concealed, but it can be folded to fit into a backpack. In the folded state, the bow cannot be loaded with arrows for quick access and shooting at an opponent, so before using it, you need to unfold it, select the type of arrow, and only then fire. After firing, the bow will reload for 2 seconds, making it not the most discreet and effective weapon. Nevertheless, the bow provides three types of arrows for use:

Standard arrow: deals 35 brute, 40 stamina damage, 50 armor penetration and knockdown the target for 2 seconds.
EMP arrow: does not deal damage but causes a brief short circuit in the surrounding area and discharges the opponent's equipment.
Repulsor arrow: pushes people and unattached objects in a 3x3 radius. People are knockdown for 6 seconds.


Thus, the compact bow offers players various tactical opportunities and ways to influence the environment, although its use requires some time and preparation, as shots require prior adjustment and bow reloading.

## Why It's Good For The Game

New item enrich the experience of players in the traitor role, providing them with more tools to accomplish their objectives. The compact bow adds variety to the choice of means for completing missions and stimulates creativity and immersion in the role-playing aspect of the game.

In the game, there is a shortage of long-range weapons available to traitors. Introducing the compact bow fills this gap, giving traitors a new option for ranged combat, thereby balancing the gameplay and providing more diversity in gameplay styles.

The new item expands the arsenal of weapons in the game, adding depth and variety to the gameplay itself. Players will have more options to choose the strategy and tactics that suit them best.

Introducing a new item can stimulate interaction between players. For example, the possibility of trading or even stealing the compact bow from other crew members can become an additional source of conflict and intrigue, making the gameplay even more exciting.

Repulsor arrows can introduce new gameplay mechanics and tactics. They can be used for creating traps, distractions, or even non-lethal takedowns, adding another layer of complexity to the traitor's toolkit and encouraging creative gameplay approaches.

:cl:
add: New traitor item: Compact Bow.
/:cl:

Xackii committed:
compactbow

Jacquerel commented:
Please provide a more detailed reason why this is good for the game than "more =  better".
Why is this specific additiona good one? What unserved niche is it filling?

Xackii commented:
> Please provide a more detailed reason why this is good for the game than "more = better". Why is this specific addition a good one? What unserved niche is it filling?
> 
> Also it'd be preferable to transcribe a lot of the detail about what the item actually does into the PR body. In-game graffiti drawn on the floor provided as a youtube video is a much poorer way of conveying basic information than just typing it.

ok

paganiy commented:
60 brute amage + 60 stamina damage + knockdown for an item that does not use cartridges, for a 11 TC it is too much. Even if it  shoots once every 2 seconds, or it can’t be put in in the backpack. Compared to the revolver for 13 TC, which requires speedloader for 2 TC + bullets requiring a lot of metal, or a speargun for 20 TC, which you need to manually recharge, and to which you have limited harpoons, the bow looks too profitable. Would be better if it dealt less damage. Maybe not even brute, but burn damage.
But I like the EMP and repuls modes.

Xackii commented:
> 60 brute damage + 60 stamina damage + knockdown for an item that does not use cartridges, for a 11 TC it is too much. Even if it shoots once every 2 seconds, or it can’t be put in in the backpack. Compared to the revolver for 13 TC, which requires speedloader for 2 TC + bullets requiring a lot of metal, or a speargun for 20 TC, which you need to manually recharge, and to which you have limited harpoons, the bow looks too profitable. Would be better if it dealt less damage. Maybe not even brute, but burn damage. But I like the EMP and repuls modes.

how much to reduce

paganiy commented:
> > 60 brute damage + 60 stamina damage + knockdown for an item that does not use cartridges, for a 11 TC it is too much. Even if it shoots once every 2 seconds, or it can’t be put in in the backpack. Compared to the revolver for 13 TC, which requires speedloader for 2 TC + bullets requiring a lot of metal, or a speargun for 20 TC, which you need to manually recharge, and to which you have limited harpoons, the bow looks too profitable. Would be better if it dealt less damage. Maybe not even brute, but burn damage. But I like the EMP and repuls modes.
> 
> how much to reduce

I think it would be balanced if it dealt about 30 damage with the same 60 stamina damage and knockdown

Xackii commented:
> > > 60 brute damage + 60 stamina damage + knockdown for an item that does not use cartridges, for a 11 TC it is too much. Even if it shoots once every 2 seconds, or it can’t be put in in the backpack. Compared to the revolver for 13 TC, which requires speedloader for 2 TC + bullets requiring a lot of metal, or a speargun for 20 TC, which you need to manually recharge, and to which you have limited harpoons, the bow looks too profitable. Would be better if it dealt less damage. Maybe not even brute, but burn damage. But I like the EMP and repuls modes.
> > 
> > 
> > how much to reduce
> 
> I think it would be balanced if it dealt about 30 damage with the same 60 stamina damage and knockdown


69 will do?

paganiy commented:
> > > > 60 brute damage + 60 stamina damage + knockdown for an item that does not use cartridges, for a 11 TC it is too much. Even if it shoots once every 2 seconds, or it can’t be put in in the backpack. Compared to the revolver for 13 TC, which requires speedloader for 2 TC + bullets requiring a lot of metal, or a speargun for 20 TC, which you need to manually recharge, and to which you have limited harpoons, the bow looks too profitable. Would be better if it dealt less damage. Maybe not even brute, but burn damage. But I like the EMP and repuls modes.
> > > 
> > > 
> > > how much to reduce
> > 
> > 
> > I think it would be balanced if it dealt about 30 damage with the same 60 stamina damage and knockdown
> 
> 69 will do?

ok, let's go

Singul0 commented:
> speargun for 20 TC, which you need to manually recharge, and to which you have limited harpoons, the bow looks too profitable. Would be better if it dealt less damage. Maybe not even brute, but burn damage. But I like the EMP and repuls modes.

and I have never seen someone buy and use a rebar crossbow ever since they were added, I'd say the syndie revolver is the golden standard, a gun that two-shot kills someone for 13TC. simple and reliable. whereas this bow is a bit more clunkier, harder and more situational to use. It still two-shots yes but the reload rate is horrible compared to the fast-shootin-tootin revolver. the only superior aspect of the bow is the knockdown. that's it. the price as-is is balanced. increasing it more will only lead to it seldom being used

moocowswag commented:
The normal bow attack itself needs to have some sort of unique mechanic in order to really stand out as a niche.

A good example would be ripping from the gunpoint damage multiplier mechanic, if you were actually stealthy it would be possible to aim at someone without them knowing and receive a very slowly growing damage multiplier the longer you were locked onto them and that lock on is broke via loss of vision on them.

But for something like that to even be viable you'd have to actually give traitors a stealth option that is both powerful enough to hide them and can be used with this.

Xackii commented:
> The normal bow attack itself needs to have some sort of unique mechanic holding it back in order to really stand out as a niche while still being allowed to be powerful.
> 
> A good example would be ripping from the gunpoint damage multiplier mechanic, if you were actually stealthy it would be possible to aim at someone without them knowing and receive a very slowly growing damage multiplier the longer you were locked onto them and that lock on is broke and reset via loss of vision on them.
> 
> But for something like that to even be viable you'd have to actually give traitors a stealth option that is both powerful enough to hide them and can be used with this.
> 
> If you dont want to do anything like that
> 
> Just increase the cooldown on arrow fabrication so you are forced to hit and run / hide.


I can make some kind of sight like a syndicate sniper or a charged shot similar in mechanics to the one made by a heretical mosquito that shoots through walls

Xackii commented:
> > The normal bow attack itself needs to have some sort of unique mechanic holding it back in order to really stand out as a niche while still being allowed to be powerful.
> > A good example would be ripping from the gunpoint damage multiplier mechanic, if you were actually stealthy it would be possible to aim at someone without them knowing and receive a very slowly growing damage multiplier the longer you were locked onto them and that lock on is broke and reset via loss of vision on them.
> > But for something like that to even be viable you'd have to actually give traitors a stealth option that is both powerful enough to hide them and can be used with this.
> > If you dont want to do anything like that
> > Just increase the cooldown on arrow fabrication so you are forced to hit and run / hide.
> 
> I can make some kind of sight like a syndicate sniper or a charged shot similar in mechanics to the one made by a heretical mosin that shoots through walls

SomeoneYouProbablyKnow commented:
On the one hand the uplink is already too full of stuff with a lot of unneeded redundancy on the other funny bow.
Consider what the snatchprod does, it's a unique craft that requires a TC (so it costs you) and some basic mats, that way you're not adding another item (one which is VERY fiddly) into the already bloated uplink and we still get the funny bow (Although in the case of the snatchprod it being kinda hidden by the crafting is the only thing saving it from being abused constantly)

Side note, do you have to select the arrow every time you fire? Because that is annoying, maybe only select the arrow on use and let it charge the standard arrow on its own? Less button presses and you still get the utilities

MrMelbert commented:
> did it

Now can you do it without chat GPT

Xackii commented:
> > did it
> 
> Now can you do it without chat GPT

if some points are not clear, I can explain them or add them to PR

EnterTheJake commented:
Waaaaaay overtuned for  11 tc , it's an assassination, anti silicon, disruption tool rolled into 1 item.

It also does not really fulfill the role of a long-range weapon since it lacks  a zoom in feature.

MrMelbert reviewed (changes requested):
Mini e-bow: 
- 10 tc
- 5 melee force
- 60 stamina damage
- 15 tox damage
- 1 second of knockdown
- 2 second CD
- fits in pockets
- no time lock / progression req

Esword
- 8 tc
- 30 melee force
- fits in pockets
- 20 minute time lock / progression req

This bow
- 11 tc
- 26 melee force (almost as strong as the esword??)
- 60 stamina damage (same as ebow)
- 60 brute damage (4x more damage than ebow
- 4 seconds of knockdown (4x duration than ebow)
- 1.5 second CD (1.3x faster than ebow)
- fits in backpack
- ALSO can EMP at range
- no time lock / progression req (you can buy this beast instantly in a round)

SomeoneYouProbablyKnow commented:
> Waaaaaay overtuned for 11 tc , it's an assassination, anti silicon, disruption tool rolled into 1 item

Idk i think 11 is fair. It doesn't do any of those that well. Guns and Stuns do better for assassination or even e-bow, emp flashlight is amazing for EMP utility ajd emp grenades destroy silicones. I feel like it's a nice backup, a fallback tool but it doesn't really do anything unique in the uplink (which is why i think it should be crafted with a tc cost instead, it and many other redundant items in the uplink)

But yea a zoom would be cool, can't shoot through walls but still

projectkepler-ru reviewed (changes requested):
You really need to add a time progression cost for this, 50-60 minutes will do.

MrMelbert commented on code/modules/projectiles/guns/ballistic/bows/bow_types.dm:
```diff
@@ -45,3 +45,130 @@
+	drawn = TRUE
+	slot_flags = null
+	w_class = WEIGHT_CLASS_NORMAL
+	force = 26
```
The other stats make sense to me but I just want to know why it's stronger than an armblade in melee

Xackii replied:
to have some opportunity to fight back when the cartridge is being reloaded.
but you're right, the number is too big

MrMelbert replied:
(IMO) The weapon shouldn't be capable of everything. If it's supremely good at range, being in melee should put you at a disadvantage. Then the player can play around the disadvantage, either by acquiring a proper melee weapon (at 11tc, you can afford an esword as well) or by playing around it (don't go into melee)

Xackii replied:
in this case, I can leave the base damage of the bow at 15

Jacquerel commented:
> You really need to add a time progression cost for this, 50-60 minutes will do.

I don't know if this was serious but our current philosophy of time locks basically mean that the only values you should really be using are:

0- Most items
30- Things you can use to rapidly depopulate the station or devastate a large area, shouldn't be available at round start
90+- Vanity items for showing off

There isn't really any point to have a setting at any of the other values.

Eronymun commented:
Very cool, I like it. I was hoping bows would end up in regular play somehow.

In particular I like the sound design, this thing just screams traitor when you pull it out.

Xackii commented:
> Mini e-bow:
> 
> * 10 tc
> * 5 melee force
> * 60 stamina damage
> * 15 tox damage
> * 1 second of knockdown
> * 2 second CD
> * fits in pockets
> * no time lock / progression req
> 
> Esword
> 
> * 8 tc
> * 30 melee force
> * fits in pockets
> * 20 minute time lock / progression req
> 
> This bow
> 
> * 11 tc
> * 26 melee force (almost as strong as the esword??)
> * 60 stamina damage (same as ebow)
> * 60 brute damage (4x more damage than ebow
> * 4 seconds of knockdown (4x duration than ebow)
> * 1.5 second CD (1.3x faster than ebow)
> * fits in backpack
> * ALSO can EMP at range
> * no time lock / progression req (you can buy this beast instantly in a round)

Esword is very strong and useful for its price because it can block melee and ranged attacks. The crossbow is also very strong because it is long-range, stuns and inflicts the toxin lychee, which is more difficult than brute and burn.

MrMelbert commented:
> Esword is very strong and useful for its price because it can block melee and ranged attacks. The crossbow is also very strong because it is long-range, stuns and inflicts the toxin lychee, which is more difficult than brute and burn.

My point being made here was that the box ( in its current state ) is better than both items combined, for about half the price

Xackii commented:
> > Esword is very strong and useful for its price because it can block melee and ranged attacks. The crossbow is also very strong because it is long-range, stuns and inflicts the toxin lychee, which is more difficult than brute and burn.
> 
> My point being made here was that the bow ( in its current state ) is better than both items combined, for about half the price. At this state it should be like... 18tc.


Purchasing an ebow and esword will make you a more formidable opponent in both close quarters and ranged combat

SomeoneYouProbablyKnow commented:
> This bow
> 
> * 11 tc
> * 26 melee force (almost as strong as the esword??)
> * 60 stamina damage (same as ebow)
> * 60 brute damage (4x more damage than ebow
> * 4 seconds of knockdown (4x duration than ebow)
> * 1.5 second CD (1.3x faster than ebow)
> * fits in backpack
> * ALSO can EMP at range
> * no time lock / progression req (you can buy this beast instantly in a round)

Why does it have 26 melee the heck? That's funny. It is kinda overtuned. Ideally it should pick a lane. 
Either a main weapon with melee damage, higher price, progression requirements etc. 
Or a utility tool that can function as a weapon. A sort of jack of all trades. But then it needs to get rid of stam damage, most knockdown and melee damage.

> Esword is very strong and useful for its price because it can block melee and ranged attacks. 

Well not really, if you want block you get desword. An esword is most commonly used as a damage stick to finish people off. 3 TC for extra utility and range damage is kinda crazy. But also if you need a damage stick you'de better off not spending too much.
Honestly don't know why e-sword even has a time requirement but then again 20 minutes is nothing.

> The crossbow is also very strong because it is long-range, stuns and inflicts the toxin lychee, which is more difficult than brute and burn.

Let's be honest the actual damage from the ebow is significantly lower since you refen toxin damage. It's enough damage to kill someone if you keep shooting at them but most of the time it's not that large. The stun is the main feature of the ebow. All the confusion effects from the toxin are a big deal though.
Now that i think about it an infinite ranged weapon is an unhead of thing for the uplink. Traitors already steal the cap laser all the time for that reason.

Honestly should testmerge this and see how people abuse it.

Sakamoto4ka commented:
Esword can be hold in second hand like a shield and have a good embed RP. Desword just have a cool block chance. I notice that desword(with noslips) often used by new players cause its simple tool to rush stunstick/laser red guys. Esword is more like choice of advanced players for different style of fight which include main weapon in main hand

Wallemations reviewed (approved):
if it's a traitor item you might wanna try red instead of green but otherwise the sprites work fine

SomeoneYouProbablyKnow commented:
> if it's a traitor item you might wanna try red instead of green but otherwise the sprites work fine

We do have confusing color coding on that front. E bow projectiles are green but making them red might be confusing since they can't go through windows like regular lasers. I assume this too works by the same rules. I'd say making both this and the ebow orange could make sense but laser carbines took that already. Green is a fine color, ninjas uses so it's antag adjacent at least.

Eronymun commented:
Oh huh yeah that melee damage is wacky. At least have it lowered to toolbox level or something, it would be more cool to see this used as a bow than an everything weapon.

Xackii committed:
balanceandwires

Xackii commented:
write everything down again

Xackii commented:
Removed the melee damage, added a sight, increased cooldown to 2 seconds, reduced damage arrow damage to 15, stamina to 40, and knockdown to 2. Added a wires panel for the bow. Three wires correspond to each type of arrow; they can be pulsed and cut. Only one type of arrow can be pulsed at a time. When an arrow is pulsed, it receives a certain buff and debuff, and the bow's reload time is increased by 0.5 seconds. For example, if the damage arrow is pulsed, it will deal 40 damage instead of 15, consume 10 stamina instead of 40, and cause 1 second of knockdown instead of 2. If an arrow is cut, it cannot be loaded, resulting in an error, but the bow's reload time is reduced by 0.5 seconds. If all three wires are cut, the bow will stop functioning. There is also a fourth wire responsible for automatic arrow loading. By default, the mod is set to manual. Pulsing it allows you to choose which type of arrow will be automatically loaded without a menu. If this wire is cut, arrows will be chosen randomly.

Eronymun commented:
The range damage feels a bit low now, so is the stam damage. Now I am worried it's non-competitive with the e-crossbow. 35 damage standard and two shot range for pulsing big damage mode feels fair. Maybe 50 st damage included when unpulsed. It does have more utility though, at least.

Xackii committed:
dmgandpene

Xackii commented:
> The range damage feels a bit low now. Now I am worried it's non-competitive with the e-crossbow as a weapon choice for damage. 35 damage standard and two shot range for pulsing big damage mode feels fair, especially if the knock down is only 1 second but recharge takes 2. The type can also be tox or mix brute and burn. Straight brute is the most common armor type in the game, so the bow is more likely to do less damage per shot (or give it full armor piercing). It does have more utility though, so maybe that is where it's niche lies.


forgot about the armor. made 35 dmg and 40 stam and 50 pene. if pulsed making 55 dmg and 20 stam

another-expler commented:
To be honest, i like more the idea of non energetic tactical composite bow.
-sight
-foldable
-0.5s to tight the string.
And it can be used as unifersal helper with many types of arrows, that come with it in something like quiver. For example:sleeping dart, damage arrow, emp arrow, maybe 
?shotgun arrow?, penetrating arrow, poison maybe, bola arrow, or i dunno, taser arrow, you can create a lot of random coming types of arrows. . The main idea is to mix different types of arrows to reach the goal.

Xackii commented:
> To be honest, i like more the idea of non energetic tactical composite bow. -sight -foldable -0.5s to tight the string. And it can be used as unifersal helper with many types of arrows, that come with it in something like quiver. For example:sleeping dart, damage arrow, emp arrow, maybe ?shotgun arrow?, penetrating arrow, poison maybe, bola arrow, or i dunno, taser arrow, you can create a lot of random coming types of arrows. . The main idea is to mix different types of arrows to reach the goal.

I was thinking about adding modules that can be bought for 2-4 tk and they will give new types of arrows, but so far I have settled on 3 types

KingkumaArt commented:
> and I have never seen someone buy and use a rebar crossbow ever since they were added, I'd say the syndie revolver is the golden standard, a gun that two-shot kills someone for 13TC. simple and reliable. whereas this bow is a bit more clunkier, harder and more situational to use. It still two-shots yes but the reload rate is horrible compared to the fast-shootin-tootin revolver. the only superior aspect of the bow is the knockdown. that's it. the price as-is is balanced. increasing it more will only lead to it seldom being used

Hey, Rebar ebow PR guy here. The rebar ebow is meant as an alpha strike/assassination weapon, where you jump them and finish the person off with other weapons if they're still kicking. The reload rate is slow compared to the revolver, but thats by design - its  meant as a stealth single target weapon, rather than the revolver, which is better for long engagements but is very loud.

Xackii commented:
> Mini e-bow:
> 
> * 10 tc
> * 5 melee force
> * 60 stamina damage
> * 15 tox damage
> * 1 second of knockdown
> * 2 second CD
> * fits in pockets
> * no time lock / progression req
> 
> Esword
> 
> * 8 tc
> * 30 melee force
> * fits in pockets
> * 20 minute time lock / progression req
> 
> This bow
> 
> * 11 tc
> * 26 melee force (almost as strong as the esword??)
> * 60 stamina damage (same as ebow)
> * 60 brute damage (4x more damage than ebow
> * 4 seconds of knockdown (4x duration than ebow)
> * 1.5 second CD (1.3x faster than ebow)
> * fits in backpack
> * ALSO can EMP at range
> * no time lock / progression req (you can buy this beast instantly in a round)

as I said 3 days ago

Xackii commented:
> Removed the melee damage, added a sight, increased cooldown to 2 seconds, reduced damage arrow damage to 35, stamina to 40, and knockdown to 2. Added a wires panel for the bow. Three wires correspond to each type of arrow; they can be pulsed and cut. Only one type of arrow can be pulsed at a time. When an arrow is pulsed, it receives a certain buff and debuff, and the bow's reload time is increased by 0.5 seconds. For example, if the damage arrow is pulsed, it will deal 40 damage instead of 15, consume 10 stamina instead of 40, and cause 1 second of knockdown instead of 2. If an arrow is cut, it cannot be loaded, resulting in an error, but the bow's reload time is reduced by 0.5 seconds. If all three wires are cut, the bow will stop functioning. There is also a fourth wire responsible for automatic arrow loading. By default, the mod is set to manual. Pulsing it allows you to choose which type of arrow will be automatically loaded without a menu. If this wire is cut, arrows will be chosen randomly.

.

sylvia-from-fulp-station commented:
Have you thought of adding a Cryptographic sequencer interaction with the wires?

Xackii commented:
> Have you thought of adding a Cryptographic sequencer interaction with the wires?


I haven't thought about it but it sounds interesting

Xackii committed:
Merge branch 'master' into compactbow

Eronymun commented:
@MrMelbert

Xackii commented:
> @MrMelbert

waiting

moocowswag commented:
As long as ebows are green i think the compact bow is okay being green, the syndicate is a group of different coorporations and one of them makes green hunting themed weapons.

SomeoneYouProbablyKnow commented:
Red projectile goes through windows
Green projectile doesn't
You can't make it red without messing up that very neat visual aspect we have

Xackii commented:
> Merge when

when will Melbert answer

MrMelbert commented:
I apologize for keeping you waiting but it was because I wasn't actually very interested in merging it. And it seems like no other maintainers are interested either. 

I prefer the simplicity of traitor items like the mini-e-bow or the e-sword. Compared to them, this item feels very overloaded. (Mini-e-bow is fire and forget. This has three arrow types and can be hacked with wires.)

Now that's not to say I don't think it could fit somewhere, I had a shower thought a few days ago that this would fit as an alternate ninja weapon? 

Ninja is all about having a ton of utility, and this bow has a ton of utility. 
Even the color matches Ninja, sort of. 
We could allow Space Ninjas to pick between Katana, Bow, or some other in-theme weapon when they spawn (Naginata or something?). Could be fun to add variety to them.

Xackii commented:
> I apologize for keeping you waiting but it was because I wasn't actually very interested in merging it. And it seems like no other maintainers are interested either.
> 
> I prefer the simplicity of traitor items like the mini-e-bow or the e-sword. Compared to them, this item feels very overloaded. (Mini-e-bow is fire and forget. This has three arrow types and can be hacked with wires.)
> 
> Now that's not to say I don't think it could fit somewhere, I had a shower thought a few days ago that this would fit as an alternate ninja weapon?
> 
> Ninja is all about having a ton of utility, and this bow has a ton of utility. Even the color matches Ninja, sort of. We could allow Space Ninjas to pick between Katana, Bow, or some other in-theme weapon when they spawn (Naginata or something?). Could be fun to add variety to them.

ok
